app.py

```diff
@@ -65,36 +72,37 @@ def scan_for_hexpansion(self): # Is there a proper way of doing this so I don't
                 # We found it, the search is over!
                 print("Found the desired hexpansion in port " + str(port))
                 return HexpansionConfig(port)
-        
+
         return None
 
     def _handle_buttondown(self, event: ButtonDownEvent):
         if BUTTON_TYPES["CANCEL"] in event.button:
             self._cleanup()
             self.minimise()
-        
+
         if BUTTON_TYPES["UP"] in event.button:
             self._cleanup()
-            self.jitter_factor += 0.1   
+            self.jitter_factor += 0.1
             if self.jitter_factor >= 1:
                 self.jitter_factor = 1.0
             self.jitter_timeout_ms = self.generate_timeout() # Update the timeout whenever the amount consumed has changed
-        
+
         if BUTTON_TYPES["DOWN"] in event.button:
             self._cleanup()
             self.jitter_factor -= 0.1
             if self.jitter_factor < 0:
                 self.jitter_factor = 0.0
             self.jitter_timeout_ms = self.generate_timeout() # Update the timeout whenever the amount consumed has changed
 
-
-
     def _cleanup(self):
         eventbus.remove(ButtonDownEvent, self._handle_buttondown, self.app)
 
     def update(self, delta):
         if self.hexpansion_config is not None:
-            self.jitter_randomly()
+            try:
```

Comment:
Wrapping this in a try catch to avoid the app crashing if you remove the hexpansion while it's jittering